src/renderer.js

```diff
@@ -59,7 +59,10 @@ const handleFileSelect = event => {
 
```

Comment:
I would await the arrayBuffer here:
`const file = await file.arrayBuffer();`

then use it in the object
`data: {
  ...
  file,
  path: file.path,
}`

Also, do you need a try-catch block for handling the possible errors ?

Reply:
yes much readable than mine, I'll update accordingly 👍🏼  and will check the try catch handler as well